### ISSUE TITLE
Enable direct access to password generator

### ISSUE THREAD

**Issue by d-faure** [Kunzisoft/KeePassDX#518]
**Is your feature request related to a problem? Please describe.**
For now (v2.5rc1), the password generator is only available when creating new entries. This would be convenient to have access to it all the time (as KeePassXC does), noticeably if/when the templating system would be implemented.

**Describe the solution you'd like**
Provide an access icon near the parameter in the title bar.

**Comment by J-Jamet** [comment]
I agree, I even thought about it and I think we need to go further and offer a separate password generation application, more advanced.
The goal is to leave a standard version in the application, lighter. And the separate app would contain the generation system by passphrase (described in the pull request #110 ) which will be heavier because it will contain the generation dictionaries.
If both applications are installed on the same device, KeePassDX recognizes it and replaces the basic generation of password.

**Comment by Ilithy** [comment]
A brilliant idea!

Bit wouldn't this create a weakness or a process could retrieve information when switching from one to the other?

Is for only to keep the core (KeepassDX) as light as possible, With the evolution of the storage of our devices, wouldn't this be useless?

If I may ask one more question, what is, for you, the acceptable weight limit for an application such as KeepassDX?

Please don't see any criticism on it, and sorry if I am disturbing this thread.

**Comment by J-Jamet** [comment]
No worries, these are legitimate questions.

> But wouldn't this create a weakness or a process could retrieve information when switching from one to the other?

If things are done right, no. It will be necessary to verify that the calling application corresponds to that which is requested. I need to study this point in more detail.  If the applications are signed and are part of the same package, there should be no security concerns.

> Is for only to keep the core (KeepassDX) as light as possible, With the evolution of the storage of our devices, wouldn't this be useless?

KeePassDX is also dedicated to older devices, the minimum SDK is 14 (Android 4.0). And adding dictionaries with thousands of words for all languages ​​will greatly increase the size of the application. I'm thinking of a plugin system for that.

> If I may ask one more question, what is, for you, the acceptable weight limit for an application such as KeepassDX?

I would like to stay around 10 MB.

**Comment by Ilithy** [comment]
My apologies for the delay of my reply;

I understand your reasoning, it is quite consistent indeed.

10Mb is really very light (not a criticism) when I compare to KP2A (long installed on my smartphone) which is more than 70Mb (without the additional modules) for 22Mb installed for KeepassDX, it clearly makes the old low storage device more easily accessible.

Thanks for your answers

**Comment by J-Jamet** [comment]
Yes when I speak of 10MB, it is to describe the APK package. The fewer libraries there are, the less the final place will be on the device. After that, it depends on the number of databases used, settings, etc ... Put I try to optimize at best.
